share/ipld/get.go

```diff
@@ -170,22 +170,24 @@ func GetLeaves(ctx context.Context,
 	wg.Wait()
 }
 
-// GetLeavesByNamespace returns leaves and corresponding proof that could be used to verify leaves
+// GetLeavesByNamespace collects leaves and corresponding proof that could be used to verify leaves
 // inclusion. It returns as many leaves from the given root with the given namespace.ID as it can
-// retrieve. If no shares are found, it returns both data and error as nil. If non-nil
-// proofContainer param passed, it will be filled with data required for inclusion verification. A
+// retrieve. If no shares are found, it returns error as nil. A
 // non-nil error means that only partial data is returned, because at least one share retrieval
 // failed. The following implementation is based on `GetShares`.
 func GetLeavesByNamespace(
 	ctx context.Context,
 	bGetter blockservice.BlockGetter,
 	root cid.Cid,
 	nID namespace.ID,
-	maxShares int,
-	proofContainer *Proof,
-) ([]ipld.Node, error) {
+	retrievedData *RetrievedData,
```

Comment:
We would avoid RetrieveData, fetchedBounds, and refactoring if we just passed options here. 

`WithProofs(proofContainer), WithData([]ipld.Node)`.

Don't feel strongly about and we can keep this as you made it, but with renaming RetrievedData to NamespaceData

Reply:
I wanted to hide all logic(with conditions `if collectProofs/collectLeaves`) inside and it now looks tidy imo. I'm fine w `NamespaceData`

Reply:
Let's name it so